### PR TITLE
Feat: Reload debug server on changes to src folder

### LINKED ISSUE #45: Debug Server doesn't track src folder
The server doesn't restart when changes to src folder happens.

### PR DESCRIPTION
**Pull request by ksgr5566**
Fixes #45. Used `watchgod` to track changes in **src** folder. Only on quart debug mode the server reloads when changes are detected.

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -9,12 +9,52 @@
...
 import time
+from watchgod import awatch 
+import asyncio
 
 load_dotenv()
 
 app = Quart(__name__)
 Compress(app)
 
+src_dir = "src"
```
Comment (ChakshuGautam) [inline api.py:20]: I would suggest we move this to watchdog.py since this file is named `api.py` so only controllers/quart related things should be here.

**Review by ChakshuGautam — APPROVED** [review]

**Comment by ChakshuGautam** [comment]
Overall LGTM. Can you also suggest alternative approaches to this? Just want to cover my bases here. I know that flask/quart also monitors folders and can restart the server when a file changes. Why does that not work in this case?

File: api.py (hunk 1 above)
Comment (ksgr5566) [reply api.py]: I have done that. Please review.

**Comment by ksgr5566** [comment]
> Overall LGTM. Can you also suggest alternative approaches to this? Just want to cover my bases here. I know that flask/quart also monitors folders and can restart the server when a file changes. Why does that not work in this case?

As you have already mentioned in #45,  the quart debug server may not be able to detect changes in the "src" folder since the imports are resolved dynamically at runtime. Other approaches that I have looked up so far are:
- Using **docker volume mounting** (mount the volumes in docker run command and updating dockerfile to set the quart command to run when the container starts)
- Using **nodemon** to watch changes in src folder.

I don't think without using `watchgod` or any other external library we can track changes in `src` folder while using `quart --app api --debug run` directly.

**Pull request merged**